overlord/devicestate/devicemgr.go

```diff
@@ -329,22 +335,74 @@ func (m *DeviceManager) maybeSetupUbuntuSave() error {
 	if !runMntSaveMounted {
 		// we don't have ubuntu-save, save will be used directly
 		logger.Noticef("no ubuntu-save mount")
-		m.saveAvailable = true
 		return nil
 	}
 
-	logger.Noticef("bind-mounting ubuntu-save under %v", dirs.SnapSaveDir)
+	// In newer core20/core22 we have a mount unit for ubuntu-save, which we
+	// we will try to start first. Invoking systemd-mount in this case would fail.
+	err = exec.Command("systemctl", "enable", "var-lib-snapd-save.mount", "--now").Run()
+	if err == nil {
+		logger.Noticef("mount unit for ubuntu-save was started")
+		return nil
+	}
```

Comment:
It would be nice to have an `else` switch here, so that if the `err` is not the one we would expect to receive when a mount unit is not present, it's reported back to the caller. In other words, if the mount unit is present but for some reason it fails to start, we should abort and report the error back to the caller, instead of attempting to manually mount the directory. Otherwise the error is lost.

I'm not sure that this is possible to do by just looking at `err`; you might need to parse the stderr too?

Reply:
Thank you for pointing this out - I implemented this suggestion and now check the error message